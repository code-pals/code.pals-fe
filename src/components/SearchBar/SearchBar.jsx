import {
  Input,
  Box,
  Stack,
  InputGroup,
  InputLeftElement,
  Icon,
  Center,
} from '@chakra-ui/react';
import { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { searchPostsAndComments } from '../../services/fetch-utils';
import CommentBox from '../CommentBox/CommentBox';
import PostHomeBox from '../PostHomeBox/PostHomeBox';

export default function SearchBar() {
  const [search, setSearch] = useState('');
  const [comments, setComments] = useState([]);
  const [posts, setPosts] = useState([]);
  const history = useHistory();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const responseArr = await searchPostsAndComments(search);
    console.log(responseArr);
    setComments(responseArr.body[0]);
    setPosts(responseArr.body[2]);
    // history.push({
    //   pathname: '/results',
    //   state: comments })
  };

  console.log(comments, posts);
  return (
    <>
      <Center>
        <Box m="20px" width="50%">
          <Stack spacing={4}>
            <form onSubmit={handleSubmit}>
              <InputGroup>
                <InputLeftElement
                  children={<Icon name="search" color="gray.300" />}
                />
                <Input
                  type="text"
                  placeholder="Search"
                  value={search}
                  onChange={(e) => setSearch(e.target.value)}
                />
              </InputGroup>
              <button type="submit">Submit</button>
            </form>
          </Stack>
        </Box>
      </Center>
      <div>
        {posts.map((post) => {
          return <PostHomeBox key={post.postId} post={post} />;
        })}

        {comments.map((comment) => {
          return <CommentBox key={comment.commentId} comment={comment} />;
        })}
      </div>
    </>
  );
}
