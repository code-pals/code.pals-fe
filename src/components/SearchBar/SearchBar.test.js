import { render } from '@testing-library/react';
import SearchBar from './SearchBar.jsx';

it('should render SearchBar', () => {
  const { container } = render(<SearchBar />);
  expect(container).toMatchSnapshot();
});
