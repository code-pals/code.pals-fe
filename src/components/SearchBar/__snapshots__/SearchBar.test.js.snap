// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar 1`] = `
<div>
  <div
    class="css-gmuwbf"
  >
    <div
      class="css-s9nw05"
    >
      <div
        class="chakra-stack css-17kdbol"
      >
        <form>
          <div
            class="chakra-input__group css-4302v8"
          >
            <div
              class="chakra-input__left-element css-o84xzr"
            >
              <svg
                class="chakra-icon css-1fdnb51"
                focusable="false"
                name="search"
                viewBox="0 0 24 24"
              >
                <g
                  stroke="currentColor"
                  stroke-width="1.5"
                >
                  <path
                    d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                    fill="none"
                    stroke-linecap="round"
                  />
                  <path
                    d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                    fill="currentColor"
                    stroke-linecap="round"
                  />
                  <circle
                    cx="12"
                    cy="12"
                    fill="none"
                    r="11.25"
                    stroke-miterlimit="10"
                  />
                </g>
              </svg>
            </div>
            <input
              class="chakra-input css-0"
              placeholder="Search"
              type="text"
              value=""
            />
          </div>
          <button
            class="chakra-button css-1gek3pj"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    class="css-0"
  >
    <div
      class="css-1mpt2dh"
    />
    <div
      class="css-gmuwbf"
    >
      <div
        class="css-0"
      >
        <br />
      </div>
    </div>
  </div>
</div>
`;
